### DISCUSSION TITLE
diracos2 installer script failed unpacking on arm64 (M2)

### DISCUSSION THREAD

**Discussion by randria** [DIRACGrid/DIRAC#7546] [Q&A] [answered]
hi
I have added a comment to this [issue#6977](https://github.com/DIRACGrid/DIRAC/issues/6977) opened since april 2023. I have failure extraction cf. below when running the diracos2 bash installer script (version 2.38) for Darwin on M2 arm running OSX Sonoma 14.1.1

```bash
Unpacking payload ...
./DIRACOS-2.38-Darwin-arm64.sh: line 265: 17637 Exit 141                extract_range $boundary1 $boundary2
     17638 Killed: 9               | "$CONDA_EXEC" constructor --extract-tarball --prefix "$PREFIX"
./DIRACOS-2.38-Darwin-arm64.sh: line 268: /Users/xxx/git/grid-on-client/diracos/preconda.tar.bz2: No such file or directory
```

Does anyone met the same error and fixed it ?
Thank for any help.
Bests,

**Comment by chrisburr** [comment]
You're probably seeing a weird code signing thing, macOS can be a bit weird.

Can you try again after rebooting?

**Reply by randria** [reply]
well sorry to say but I already did many reboot sequence and another now. And yet it still yells no `preconda.tar.bz2` generated.

**Reply by chrisburr** [reply]
The `Killed: 9` is the problem, likely from macOS itself. Can you try running the installer with `bash -x ....`. Hopefully you'll then be able to see which command is being killed.

**Reply by randria** [reply]
seems to be `conda.exe` after being extracted. Because I run just the binary it's killed.
```bash
zsh: killed     /Users/xxx/git/grid-on-client/diracos/conda.exe
```

**Reply by randria** [reply]
maybe I can build a local conda constructor

**Reply by randria** [reply]
what is `conda.exe` extracted ? because I have installed a conda constructor and tryed to replace `conda.exe` and execute juste `constructor` but it yells option `--extract-tarball` doesn't exist so I guess it's wrapping something else but not constructor
```
CONDA_EXEC="$PREFIX/conda.exe"
...
"$CONDA_EXEC" constructor --prefix "$PREFIX" --extract-tarball < "$PRECONDA"
```

**Reply by randria** [reply, marked as answer]
well I fixed something by editing the miniforge3 installation script and adding in `condarc` channels `diracgrid` and seems to work now.

**Reply by randria** [reply]
Yet it's problematic because it works when I build with the constructor conda step-by-step but it failed when I fetched the built version releases here https://github.com/DIRACGrid/DIRACOS2/releases . 

I have tested versions `2.38 2.37 2.36` and got the same error message. reboot or kill processes don't change anything
```bash
./DIRACOS-2.36-Darwin-arm64.sh: line 265: 42744 Exit 141                extract_range $boundary1 $boundary2
     42745 Killed: 9               | "$CONDA_EXEC" constructor --extract-tarball --prefix "$PREFIX"
./DIRACOS-2.36-Darwin-arm64.sh: line 268: /Users/xxx/diracos/preconda.tar.bz2: No such file or directory
```

**Reply by JackFannon** [reply]
@randria I'm running into the same issue that you were with an M1 Mac. Do you have any more information on the fix you implemented?
